core/src/main/clojure/xtdb/metrics.clj

```diff
@@ -0,0 +1,71 @@
+(ns xtdb.metrics
+  (:require [clojure.tools.logging :as log]
+            [juxt.clojars-mirrors.integrant.core :as ig]
+            [xtdb.error :as err])
+  (:import (com.sun.net.httpserver HttpServer HttpHandler)
+           (java.net InetSocketAddress)
+           (io.micrometer.core.instrument MeterRegistry Meter Measurement Timer Gauge Tag Counter)
+           (io.micrometer.core.instrument.binder MeterBinder)
+           (io.micrometer.core.instrument.binder.jvm ClassLoaderMetrics JvmMemoryMetrics JvmHeapPressureMetrics JvmGcMetrics JvmThreadMetrics)
+           (io.micrometer.core.instrument.binder.system ProcessorMetrics)
+           (io.micrometer.core.instrument.simple SimpleMeterRegistry)
+           (io.micrometer.prometheus PrometheusMeterRegistry PrometheusConfig)
+           (java.time Duration)))
+
+(defn meter-reg ^MeterRegistry
+  ([] (meter-reg (SimpleMeterRegistry.)))
+  ([meter-reg]
+   (let [bind-metrics (fn [& metrics] (run! #(.bindTo ^MeterBinder % meter-reg) metrics))]
+     (bind-metrics
+      (ClassLoaderMetrics.)
+      (JvmMemoryMetrics.)
+      (JvmHeapPressureMetrics.)
+      (JvmGcMetrics.)
+      (ProcessorMetrics.)
+      (JvmThreadMetrics.))
+     meter-reg)))
+
+(defmethod ig/init-key :xtdb/meter-registry [_ {}]
+  (meter-reg (PrometheusMeterRegistry. PrometheusConfig/DEFAULT)))
+
+(defmethod ig/halt-key! :xtdb/meter-registry [_ ^MeterRegistry registry]
+  (.close registry))
+
+(defn add-counter [reg name {:keys [description]}]
+  (cond-> (Counter/builder name)
+    description (.description description)
+    :always (.register reg)))
+
+(def percentiles [0.75 0.85 0.95 0.98 0.99 0.999])
+
+(defn add-timer [reg name {:keys [^String description]}]
+  (cond-> (.. (Timer/builder name)
+              (minimumExpectedValue (Duration/ofNanos 1))
```

Comment:
do we need to specify these? I think any value we put here will be both arbitrary and likely inaccurate, so would err on the side of not providing them at all unless it has other impacts I'm not aware of